frame/nomination-pools/src/lib.rs

```diff
@@ -582,25 +583,259 @@ pub struct PoolRoles<AccountId> {
+	/// commission is zero, `None` will be inserted and `payee` will be ignored.
+	fn maybe_update_current(
+		&mut self,
+		commission: &Perbill,
+		payee: T::AccountId,
+	) -> DispatchResult {
+		ensure!(!self.throttling(&commission), Error::<T>::CommissionChangeThrottled);
+		ensure!(
+			GlobalMaxCommission::<T>::get().map_or(true, |m| commission <= &m),
+			Error::<T>::GlobalMaxCommissionExceeded
+		);
+		ensure!(self.max.map_or(true, |m| commission <= &m), Error::<T>::CommissionExceedsMaximum);
+
+		self.current = Some((*commission, payee));
+
+		// update `throttle.previous_set_at` if a throttle has been configured.
+		let _ = self
+			.throttle
+			.as_mut()
+			.map(|t| t.register_change(<frame_system::Pallet<T>>::block_number()));
+		Ok(())
+	}
+
+	/// Set the pool's maximum commission.
+	///
+	/// The pool's maximum commission can initially be set to any value, below
+	/// global maximum if set, and only smaller values thereafter. If larger
+	/// values are attempted, this function will return a dispatch error.
+	///
+	/// If `current.0` is larger than an updated max commission value, then
+	/// `current.0` will also be updated to the new maximum.
+	fn maybe_update_max(&mut self, new_max: Perbill) -> DispatchResult {
+		ensure!(
+			GlobalMaxCommission::<T>::get().map_or(true, |m| new_max <= m),
+			Error::<T>::GlobalMaxCommissionExceeded
+		);
+		if let Some(old) = self.max.as_mut() {
+			if new_max > *old {
+				return Err(Error::<T>::MaxCommissionRestricted.into())
+			}
+			*old = new_max;
+		} else {
+			self.max = Some(new_max)
+		};
+		let _ = self.current.as_mut().map(|(x, _)| *x = (*x).min(new_max));
+		Ok(())
+	}
+
+	/// Set the pool's commission throttle settings.
+	///
+	/// Once a throttle configuration has been set, only more restrictive values
+	/// can be set thereafter. These restrictions translate to increased
+	/// `min_delay` values and decreased `max_increase` values.
+	fn maybe_update_throttle(
+		&mut self,
+		change_rate: CommissionChangeRate<T::BlockNumber>,
+	) -> DispatchResult {
+		ensure!(
+			&self.throttle.as_ref().map_or(true, |t| t.less_restrictive(&change_rate)),
+			Error::<T>::CommissionThrottleNotAllowed
+		);
+
+		self.throttle = Some(CommissionThrottle {
+			change_rate,
+			previous_set_at: self.throttle.as_ref().map_or(None, |t| t.previous_set_at),
+		});
+
+		Ok(())
+	}
+
+	/// Gets the current commission (if any) and payee to be paid.
+	///
+	/// A zero commission along with a `None` payee is returned in the event a
+	/// commission has not been configured to the pool.
+	fn get_commission_and_payee(
+		&self,
+		pending_rewards: &BalanceOf<T>,
+	) -> Option<(BalanceOf<T>, T::AccountId)> {
+		self.current
+			.as_ref()
+			.map(|(commission, payee)| (*commission * *pending_rewards, payee.clone()))
+			.or(None)
+	}
+
+	/// Calculate the amount of intervals that consume a period of blocks.
+	///
+	/// The final number of intervals is returned.
+	pub fn intervals_since_block(
```

Comment:
I think this is too specific to the logic of throttle and can be one-time calculation on that function. And, nonetheless, it needs not `&self` and should be a type function.

Reply:
Moved logic into caller function.